File: TrelloApi/TrelloApi/src/Domain/Entities/UserCard.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace TrelloApi.Domain.Entities;

[Table("UserCard")]
public class UserCard
{
    [ForeignKey("User")]
    public int UserId { get; set; }
    public User User { get; set; }
    
    [ForeignKey("Card")]
    public int CardId { get; set; }
    public Card Card { get; set; }

    public UserCard(int userId, int cardId)
    {
        UserId = userId;
        CardId = cardId;
    }
}

File: TrelloApp/src/app/core/models/user-card.ts
import {Card} from './card';
import {User} from './user';

export interface UserCard {
  id: number,
  createdAt: string,
  updatedAt?: string,
  cardId: number,
  userId: number,
  card: Card,
  user: User
}

export interface AddUserCard {
  userId: number,
}
